server/src/main/java/org/elasticsearch/http/AbstractHttpServerTransport.java

```diff
@@ -281,7 +281,8 @@ protected void doStop() {
                 logger.warn(format("timed out while waiting [%d]ms for clients to close connections", shutdownGracePeriodMillis));
             }
         } else {
-            logger.debug("closing all client connections immediately");
+            logger.debug("waiting indefinitely for clients to close connections");
+            FutureUtils.get(allClientsClosedListener);
```

Comment:
IMO it'd be slightly more operator-friendly to make this a loop of timed waits, logging and retrying on timeout, so that we can have some evidence in the logs that this is where we're waiting.

Reply:
I agree on Dave's point, when troubleshooting what's going here logs of periodic timeouts would help greatly

Reply:
I added logging. I put it at info level. There's a setting along with it, the poll time for how often we log.